src/connection.ts

```diff
@@ -361,6 +390,44 @@ export class Connection extends JSForceConnection {
     }
     return result.records[0];
   }
+
+  private async loadInstanceApiVersion(): Promise<Optional<string | null>> {
+    const authFileFields = this.options.authInfo.getFields();
+    const lastChecked = authFileFields.instanceApiVersionLastRetrieved;
+    let version = getString(authFileFields, 'instanceApiVersion');
+
+    // Grab the latest api version from the server and cache it in the auth file
+    const useLatest = async () => {
+      // verifies DNS
+      await this.useLatestApiVersion();
+      version = this.getApiVersion();
+      this.options.authInfo.save({
+        instanceApiVersion: version,
+        instanceApiVersionLastRetrieved: Date.now(),
+      });
+    };
+
+    const ignoreCache = env.getBoolean('SFDX_IGNORE_API_VERSION_CACHE', false);
+    if (lastChecked && !ignoreCache) {
+      const now = Date.now();
+      const has24HoursPastSinceLastCheck = now - lastChecked > Duration.hours(24).milliseconds;
```

Comment:
Would be reasonable to establish an env var or config setting to allow customer to control the TTL, with a default of 24 hours?

Reply:
I'm open if you think that will be useful. I don't feel like it will be that useful. It is a quick call.

Reply:
I'll leave this for another item if we decide we want to offer the option.